query_engine/src/encoding.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 The CeresDB Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt, sync::Arc};
+
+use prost::bytes::Bytes;
+
+use crate::{error::Result, physical_planner::PhysicalPlanPtr};
+
+pub trait PhysicalPlanEncoder: fmt::Debug + Send + Sync + 'static {
```

Comment:
Shall we reuse `Encoder/Decoder`?

Reply:
Encoder/Decoder is not trait object safe, I have no thought about using it here.